test/conformance/helpers/tracing/traces.go

```diff
@@ -62,29 +64,84 @@ func (t SpanTree) toTestSpanTreeHelper() TestSpanTree {
 	for i := range t.Children {
 		children[i] = t.Children[i].toTestSpanTreeHelper()
 	}
-	return TestSpanTree{
+	tst := TestSpanTree{
 		Kind:                     t.Span.Kind,
 		LocalEndpointServiceName: name,
 		Tags:                     t.Span.Tags,
 		Children:                 children,
 	}
+	tst.SortChildren()
+	return tst
 }
 
 // TestSpanTree is the expected version of SpanTree used for assertions in testing.
+//
+// The JSON names of the fields are weird because we want a specific order when pretty printing
+// JSON. The JSON will be printed in alphabetical order, so we are imposing a certain order by
+// prefixing the keys with a specific letter. The letter has no mean other than ordering.
 type TestSpanTree struct {
-	Root                     bool
-	Kind                     model.Kind
-	LocalEndpointServiceName string
-	Tags                     map[string]string
+	Note                     string            `json:"a_Note,omitempty"`
+	Root                     bool              `json:"b_Root,omitempty"`
+	Kind                     model.Kind        `json:"c_Kind,omitempty"`
+	LocalEndpointServiceName string            `json:"d_Name,omitempty"`
+	Tags                     map[string]string `json:"e_Tags,omitempty"`
 
-	Children []TestSpanTree
+	Children []TestSpanTree `json:"z_Children,omitempty"`
 }
 
 func (t TestSpanTree) String() string {
 	b, _ := json.Marshal(t)
 	return string(b)
 }
 
+// SortChildren attempts to sort the children of this TestSpanTree. The children are siblings, order
+// does not actually matter. TestSpanTree.Matches() correctly handles this, by matching in any
+// order. SortChildren() is most useful before JSON pretty printing the structure and comparing
+// manually.
+//
+// The order it uses:
+//   1. Shorter children first.
```

Comment:
**Lower** children first?

Reply:
Shorter. Where height is defined as the maximal number of descendants before hitting a leaf node. For example:

```
    A
  /   \
B      C
|
D
```

'A' has a height of 2. 'B' has a height of 1. 'C' and 'D' have heights of 0.